Automattic/wp-calypso
# Modularise billing transactions state

This PR is one of many working on modularising state in Calypso. This one handles billing transactions.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42425.

#### Changes proposed in this Pull Request

* Modularise billing transactions state
* Move `ui/billing-transactions` under the main billing transactions state, as there's no point separating it.
* Update all code and tests accordingly

#### Testing instructions

* Install [Redux DevTools](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en).
* Open DevTools and switch to the Redux tab.
* Open `/read` on the live branch or your local copy of this branch.
* Click `State` on the top right to ensure you see all state, and not just the diff for the last action: 
![image](https://user-images.githubusercontent.com/409615/73774406-d3e87d80-477b-11ea-9f59-3e42cc00101f.png)
* Analyse the tree and note that there's no `billingTransactions` key, even if you expand the list of keys. This indicates that the billing transactions state hasn't been loaded yet.
* Click your avatar on the masterbar, followed by `Manage Purchases` on the sidebar.
* Verify that a `billingTransactions` key is added with the billing transactions state. This indicates that the billing transactions state has now been loaded (albeit likely empty at first).
* Click the `Billing History` tab and verify that billing transaction-related functionality works normally. This indicates that I didn't mess up.
